src/chunk_constraint.c

```diff
@@ -444,6 +444,63 @@ typedef struct ChunkConstraintScanData
 	DimensionSlice *slice;
 } ChunkConstraintScanData;
 
+/*
+ * Scan for all chunk constraints that match the given slice ID. The chunk
+ * constraints are not saved, only the matching chunk id is returned.
```

Comment:
Suggest expanding on motivation why only chunk ID is returned here.

Reply:
I removed this function entirely (embedded into caller) and applied changes based on your comments below.